src/server.rs

```diff
@@ -217,10 +231,17 @@ where
     pub fn try_bind_ephemeral(
         self,
         addr: impl Into<SocketAddr> + 'static,
-    ) -> Result<(SocketAddr, impl Future<Item = (), Error = ()> + 'static), hyper::error::Error> {
+    ) -> Result<(SocketAddr, impl Future<Output = ()> + 'static), hyper::error::Error>
+    {
         let addr = addr.into();
         let (addr, srv) = try_bind!(self, &addr)?;
-        Ok((addr, srv.map_err(|e| error!("server error: {}", e))))
+        let srv = srv.map(|result| {
+            if let Err(err) = result {
+                log::error!("server error: {}", err)
+            }
+        });
```

Comment:
This hides the error on errors, while previously we used to return a `Result` with `Err = ()`. That's because old futures are equivalent to `Future<Output = Result<T, E>>`. With this, the server future is replaced with one that immediately yields when failing to bind which is probably not what we want.

Reply:
I made a mistake. Binding is still synchronous through hyper and isn't affected by this changed!